assets/scripts/dialogs/GeotagDialog.jsx

```diff
@@ -285,104 +277,69 @@ class GeotagDialog extends React.Component {
     )
   }
 
-  render () {
-    const tileUrl = this.props.dpi > 1 ? MAP_TILES_2X : MAP_TILES
-
-    return (
-      <Dialog>
-        {(closeDialog) => (
-          <div className="geotag-dialog">
-            {!this.state.geocodeAvailable && (
-              <div className="geotag-error-banner">
-                <FormattedMessage
-                  id="dialogs.geotag.geotag-unavailable"
-                  defaultMessage="Geocoding services are currently unavailable. You can view the map,
-                    but you won’t be able to change this street’s location."
-                />
-              </div>
-            )}
-            {this.state.geocodeAvailable && (
-              <div className="geotag-input-container">
-                <GeoSearch
-                  setSearchResults={this.setSearchResults}
-                  focus={this.state.mapCenter}
-                />
-              </div>
-            )}
-            <Map
-              center={this.state.mapCenter}
-              zoomControl={false}
-              zoom={this.state.zoom}
-              onClick={this.handleMapClick}
-              useFlyTo={true}
-            >
-              <TileLayer attribution={MAP_ATTRIBUTION} url={tileUrl} />
-              <ZoomControl
-                zoomInTitle={this.props.intl.formatMessage({
-                  id: 'dialogs.geotag.zoom-in',
-                  defaultMessage: 'Zoom in'
-                })}
-                zoomOutTitle={this.props.intl.formatMessage({
-                  id: 'dialogs.geotag.zoom-out',
-                  defaultMessage: 'Zoom out'
-                })}
+  return (
+    <Dialog>
+      {(closeDialog) => (
+        <div className="geotag-dialog">
+          {geocodeAvailable ? (
+            <div className="geotag-input-container">
+              <GeoSearch
+                setSearchResults={setSearchResults}
+                focus={mapCenter}
               />
+            </div>
+          ) : (
+            <ErrorBanner />
+          )}
+          <Map
+            center={mapCenter}
+            zoomControl={false}
+            zoom={zoom}
+            onClick={geocodeAvailable ? handleMapClick : null}
+            useFlyTo={true}
+            id="map"
```

Comment:
I just added it to make the cypress test a bit leaner, but I could remove it if you'd prefer.

Reply:
Ahh, I see. I don't like to add `id`s unless JavaScript really needs to get to it (and in React, you can usually achieve that with refs). Testing introduces a whole different universe of being able to get at individual elements, and I think the thing to do is to follow `testing-library`'s approach to selecting elements: https://kentcdodds.com/blog/making-your-ui-tests-resilient-to-change

According to that philosophy, if absolutely needed, the way we can say "we need to identify this element by an ID but only for a test" is to use the `data-testid` attribute. One thing that's annoying about it is that it's present in the render of the component (but [there's a plugin to remove those in production](https://www.npmjs.com/package/babel-plugin-react-remove-properties)). What I like is that it avoids the confusion of wondering whether an `id` attribute is required because production JavaScript code is looking for it.

Reply:
Ok, I took it out!